src/Blockcore.Indexer.Core/Storage/Mongo/BlockRewindOperation.cs

```diff
@@ -0,0 +1,187 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Blockcore.Indexer.Core.Storage.Mongo.Types;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace Blockcore.Indexer.Core.Storage.Mongo;
+
+public static class BlockRewindOperation
+{
+   public static async Task RewindBlockOnIbdAsync(this MongoData storage, long blockIndex)
+   {
+      await DeleteBlockInCollectionFromTopOfTable(storage.UnspentOutputTable, nameof(UnspentOutputTable.BlockIndex),blockIndex);
+
+      await RewindInputDataIntoUnspentTransactionTableAsync(storage, blockIndex);
+
+      var output =
+         DeleteBlockInCollectionFromTopOfTable(storage.OutputTable, nameof(OutputTable.BlockIndex),blockIndex);
+
+      var input =
+         DeleteBlockInCollectionFromTopOfTable(storage.InputTable, nameof(InputTable.BlockIndex),blockIndex);
+
+      var transactions =
+         DeleteBlockInCollectionFromTopOfTable(storage.TransactionBlockTable, nameof(TransactionBlockTable.BlockIndex),blockIndex);
+
+      var addressComputed =
+         DeleteBlockInCollectionFromTopOfTable(storage.AddressComputedTable, nameof(AddressComputedTable.ComputedBlockIndex),blockIndex);
+
+      var addressHistoryComputed =
+         DeleteBlockInCollectionFromTopOfTable(storage.AddressHistoryComputedTable, nameof(AddressHistoryComputedTable.BlockIndex),blockIndex);
+
+      var addressUtxoComputed =
+         DeleteBlockInCollectionFromTopOfTable(storage.AddressUtxoComputedTable, nameof(AddressUtxoComputedTable.BlockIndex),blockIndex);
+
+
+      await Task.WhenAll(input, output, transactions, addressComputed, addressHistoryComputed, addressUtxoComputed);
+   }
+
+   public static async Task RewindBlockAsync(this MongoData storage, long blockIndex)
+   {
+      FilterDefinition<OutputTable> outputFilter =
+         Builders<OutputTable>.Filter.Eq(addr => addr.BlockIndex, blockIndex);
+      Task<DeleteResult> output = storage.OutputTable.DeleteManyAsync(outputFilter);
+
+      // delete the transaction
+      FilterDefinition<TransactionBlockTable> transactionFilter =
+         Builders<TransactionBlockTable>.Filter.Eq(info => info.BlockIndex, blockIndex);
+      Task<DeleteResult> transactions = storage.TransactionBlockTable.DeleteManyAsync(transactionFilter);
+
+      // delete computed
+      FilterDefinition<AddressComputedTable> addrCompFilter =
+         Builders<AddressComputedTable>.Filter.Eq(addr => addr.ComputedBlockIndex, blockIndex);
+      Task<DeleteResult> addressComputed = storage.AddressComputedTable.DeleteManyAsync(addrCompFilter);
+
+      // delete computed history
+      FilterDefinition<AddressHistoryComputedTable> addrCompHistFilter =
+         Builders<AddressHistoryComputedTable>.Filter.Eq(addr => addr.BlockIndex, blockIndex);
+      Task<DeleteResult> addressHistoryComputed = storage.AddressHistoryComputedTable.DeleteManyAsync(addrCompHistFilter);
+
+      // delete computed utxo
+      FilterDefinition<AddressUtxoComputedTable> addrCompUtxoFilter =
```

Comment:
We wont need this table anymore right? I guess there will be a separate PR to delete it?